### PR TITLE
ID-1330 Admin Endpoints for Orchestration

### PR DESCRIPTION
**Pull request by tlangs**
Jira: https://broadworkbench.atlassian.net/browse/ID-1330

What:

Some Admin endpoints for Orchestration so that it can get Terra Users from NIH Usernames as a part of the NBCIAccess Allowlist Sync process.

Why:

This enables Orchestration to talk to ECM instead of Thurloe to get Terra Users who have a linked NIH Account

How:

The accounts that are allowed to call the admin endpoints are specified in the config. Every request to the Admin endpoints is checked against that list of email addresses to make sure the caller is authorized.

### CODE REVIEW

File: service/src/main/java/bio/terra/externalcreds/controllers/AdminApiController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  @Override
+  public ResponseEntity<AdminLinkInfo> getLinkedAccountForExternalId(
+      Provider provider, String externalId) {
+    requireAdmin();
+    var linkedAccount = linkedAccountService.getLinkedAccountForExternalId(provider, externalId);
+    return ResponseEntity.of(linkedAccount.map(OpenApiConverters.Output::convertAdmin));
+  }
+
+  private void requireAdmin() {
+    var samUser = samUserFactory.from(request);
```
Comment (Ghost-in-a-Jar) [inline service/src/main/java/bio/terra/externalcreds/controllers/AdminApiController.java:41]: does this actually reach out to sam or just parse the request for sam user info?

File: service/src/main/resources/db/changelog/changesets/20240711_add_nih_account_table.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                  type: timestamp
+                  constraints:
+                    nullable: false
+        - createIndex:
+            columns:
+              - column:
+                  descending: true
+                  name: nih_username
+            indexName: idx_nih_username
+            tableName: nih_account
```
Comment (Ghost-in-a-Jar) [inline service/src/main/resources/db/changelog/changesets/20240711_add_nih_account_table.yaml]: It looks like this table isnt being used in this pr (linked_account instead), why are we making this new table then?

File: service/src/main/java/bio/terra/externalcreds/controllers/AdminApiController.java (hunk 1 above)
Comment (tlangs) [reply service/src/main/java/bio/terra/externalcreds/controllers/AdminApiController.java:41]: It gets the Sam user using the token in the request.

File: service/src/main/resources/db/changelog/changesets/20240711_add_nih_account_table.yaml (hunk 2 above)
Comment (tlangs) [reply service/src/main/resources/db/changelog/changesets/20240711_add_nih_account_table.yaml]: Whoops, bad rebase.

**Review by Ghost-in-a-Jar — APPROVED** [review]

**Pull request merged**